### DISCUSSION TITLE
S3 backfill without mimirtool still supported?

### DISCUSSION THREAD

**Discussion by Packetslave** [grafana/mimir#3314] [Help and support]
I note that between the 2.2 and 2.3 releases, the section in the Thanos / Prometheus migration documentation about uploading raw historical TSDB blocks directly to Mimir's storage bucket was removed.  I'm curious if this is because it is no longer a viable backfill option, or for another reason?

Context:  we're in the midst of a migration from Thanos to Mimir, and have 100+ TB of historical data sitting in S3. At that scale, the idea of copying bucket-to-bucket is significantly more attractive than a workflow of "download block from S3, call mimirtool to upload it to the Mimir compactor, repeat"

Assuming we take into account all of the caveats in the docs about working with Thanos blocks, is a direct S3-to-S3 backfill still a viable option with Mimir 2.3.1?

**Comment by pstibrany** [comment]
> Assuming we take into account all of the caveats in the docs about working with Thanos blocks, is a direct S3-to-S3 backfill still a viable option with Mimir 2.3.1?

Yes, it still works. We removed old documentation because using `mimirtool` should be easier option for most people, and can be used by end users who may not necessarily have access to the bucket. However direct upload to the bucket still works.

For Thanos-specific blocks, we have this PR open with updated documentation about how to handle Thanos blocks: https://github.com/grafana/mimir/pull/3290. Instructions there are pretty complete and you may find it useful ... or find mistakes in it, review from someone who actually performed the same migration would be appreciated!
